src/renderer/frames/cluster-frame/init-cluster-frame/init-cluster-frame.injectable.ts

```diff
@@ -20,6 +21,14 @@ const initClusterFrameInjectable = getInjectable({
 
     assert(hostedCluster, "This can only be injected within a cluster frame");
 
+    /**
+     * This is injected here to initialize it for the side effect.
+     *
+     * The side effect CANNOT be within `apiManagerInjectable` itself since that causes circular
+     * dependencies with the current need for legacy di use
+     */
```

Comment:
Interesting. Do you consider this the optimal solution, or could the dependency tree be reorganised for the good of the land?

Reply:
I was thinking more of stripping responsibilities from god-classes, such as `ApiManager`, to get rid of the cycles. This is probably not feasible right now, but then again maybe there's a chance. 

Resolve on will.